hack/lib/version.sh

```diff
@@ -66,7 +66,19 @@ kube::version::get_version_vars() {
     fi
 
     # Use git describe to find the version based on tags.
-    if [[ -n ${KUBE_GIT_VERSION-} ]] || KUBE_GIT_VERSION=$("${git[@]}" describe --tags --match='v*' --abbrev=14 "${KUBE_GIT_COMMIT}^{commit}" 2>/dev/null); then
+    if [[ -n ${KUBE_GIT_VERSION-} ]]; then
+      true
+    else
+      KUBE_GIT_VERSION=$("${git[@]}" describe --tags --match='v*' --abbrev=14 "${KUBE_GIT_COMMIT}^{commit}" 2>/dev/null) || true
+      # Fallback to /build/build-image/cross/VERSION if git describe fails
+      version_file="${KUBE_ROOT}/build/build-image/cross/VERSION"
```

Comment:
/hold
https://github.com/kubernetes/kubernetes/issues/131629#issuecomment-2856586012

This is not correct